[4.0][WIP] BS5 JS

dgrammatiko opened the pull request:
Pull Request for Issue # .

### Summary of Changes

This PR changes the JS to Bootstrap V5. Although it's a work in progress after https://github.com/joomla/joomla-cms/pull/31990/commits/d96ee241db98e6f943a0c6f931e440590eb15d00 all the components are functional so **help by testing and pointing out what is BROKEN** will be invaluable. Thanks!

### Testing Instructions
Check everything backend/front end



### What's broken:
- [x] [FIXED] Field Modals  (the extra buttons for editing/creating new Item in various forms, Modals are mostly ok)
- [x]  Side by Side Editing com Associations
- [x]  The Toolbar dropdown buttons have wrong html the dropdown needs to be the next sibling element to the button (who did this? )
- [x] [FIXED] Also Multilingual Status broken (empty popup)
- [x]  Frontend Editing
- [x] Please fix Copy Template, Manage Folders and New File modals under System>SiteTemplates>Cassiopeia Details and Files.
- [x]  Other things that I haven't tested yet

### Documentation Changes Required
There is a component that validates what's working at https://github.com/dgrammatiko/bs5/blob/main/BS5Tests.zip This can act as your DEV documentation (also it can act as part of the e2e tests)...

dgrammatiko committed:
bs5 js

dgrammatiko committed:
bs5 js

dgrammatiko committed:
Init

dgrammatiko committed:
Init

dgrammatiko committed:
Merge remote-tracking branch 'Joomla/4.0-dev' into 4.0-dev__BS5

# Conflicts:
#	administrator/components/com_banners/tmpl/tracks/default.php
#	layouts/joomla/toolbar/modal.php

chnnst commented:
dgrammatiko thank you

ceford commented:
This seems a bit short on explanation to be in the Patchtester!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31990">issues.joomla.org/tracker/joomla-cms/31990</a>.</sub>

brianteeman commented:
It's clearly labelled as a work in progress

gostn commented:
@ceford 
![image](https://user-images.githubusercontent.com/71960554/104171363-cf542c80-5402-11eb-92b1-1a335fdad862.png)

wilsonge commented on package.json:
```diff
@@ -56,6 +56,7 @@
     "short-and-sweet": "^1.0.2",
     "skipto": "^2.1.1",
     "tinymce": "^5.6.1",
+    "ttc-bs5": "^1.0.1",
```
I'll sucker up and ask the question - why your custom build of BS?

dgrammatiko replied:
Basically I didnt want to swap the bootstrap from 4 to 5 in the main repo. Probably you can have a dependency on the same package with different versions but I have no clue how to do that and it was way easier to work on clean slate than messing around with the rest of the build tools in the main repo. I guess if both JS and CSS are moving to v5 then you can import that and move the building process in the main repo (1 file). FWIW this is just me showcasing that it's not either hard or requiring a month or so to be done no strings attached here (or to put it in another way: I don't mind if the PD closes this PR, if it is already decided to roll with v4, but would be nice to not drag this too long, eg me spending time here instead of finishing the child templates)

wilsonge committed:
Move from bs4 to bs5 and half-attempt to migrate the SCSS

wilsonge commented:
![image](https://user-images.githubusercontent.com/1986000/104229968-dc920b00-5444-11eb-91d8-2ad14bbb55ed.png)
![image](https://user-images.githubusercontent.com/1986000/104229996-e4ea4600-5444-11eb-925d-0146921ee903.png)
![image](https://user-images.githubusercontent.com/1986000/104230035-ef0c4480-5444-11eb-9001-d4071e619021.png)

So I just tried a partial migration of the SCSS given the popularity of the origin discussion - branch here: https://github.com/wilsonge/joomla-cms/tree/bs5

So far it's taken over 2 hours to just get the scss to compile with all the deprecations and I have something looking like the above. It's going to take months to migrate things. Various form CSS classes have changed which means it's not just months of core development work. It's additional months after that for extension developers who have worked to get their extensions beta ready to adapt. It also means it's now or never for the 4.x Joomla series.

This is going to be discussed in the production meeting tomorrow. I think it's very very unlikely that we'll use bs5 though

dgrammatiko commented:
> It's going to take months to migrate things

@wilsonge just `@extend` the replaced classes to bring back the dropped ones. Mark them as deprecated and give it a year for both the Core to adjust but also the Devs (or for the backend keep them here tor all the v4 lifetime). Don't patch (for the time being) the markup... That was my take on the issue, not pure or optimal but totally doable, in few days (sorry can't help here I've dropped SCSS for some time now)

wilsonge commented:
It's not just the custom select classes - that's just what took most the work for the SCSS to compile - as you say for extension devs I can backport. I improved it again just now by removing the prefix BS was adding to the CSS Vars which adds back most colours. I can probably even fix the fact the entire admin now thinks it's in perpetual mobile view without screwing things up super hard.

But all the elements still don't fit padding/margin wise. For example:

![image](https://user-images.githubusercontent.com/1986000/104232334-1d3f5380-5448-11eb-8394-823c6913df1f.png)

Nothing that can't be fixed. But doing it on every page all over the CMS and for the extension devs isn't that straightforward.

wilsonge committed:
Fix CSS Vars prefix

dgrammatiko commented:
It's the renaming of the padding, margin, etc. Basically the old classes now reflect to the browsers own default styling (the css is referring to other classnames). Also I think they dropped completely the inline forms (wise choice as it's something that's against any sensible best practices) and all forms are basically inline since the multi column... Good times...

wilsonge commented:
I mean it’s the padding and margins things that will screw up most extension devs. Like it’s not forever and ever but it means going back through every page. It also invalidates any templates people have made compatible already

dgrammatiko commented:
I'll repeat myself, use extend to bring back the missing classes. Thus not breaking anything and people that want BS5 will have it. The extra baggage can be dropped in a given future date (in 2 years maybe there's a BS6 or BS is totally irrelevant, who knows). @ciar4n what do you think about all these?

ciar4n commented:
Personally, I would sooner put the work in now and make the jump to BS5. I converted the core views to BS4 for J4 originally and I am pretty sure it only took me an hour or two each day for about a week. 

BS5 is a very viable excuse to push the release date out a couple of months.

ciar4n commented:
Extending classes is a bit messy. My preference would be update the views.

dgrammatiko commented:
> My preference would be update the views.

My preference is NO bootstrap at all 🤣

brianteeman commented:
isnt the original discussion about bs5 for the site template
everything @wilsonge has done is for the admin

wilsonge committed:
Changes to make styling match

wilsonge commented:
My stuff covered all 3 clients - installation, admin + site. I just picked screenshots from installation + admin because they looked more broken visually because I can't actually install sample data to show frontend views easily

wilsonge committed:
Breakpoints and utils restored

wilsonge committed:
Alias custom-select class

wilsonge committed:
More BS4 compat + remove input-group-append

dgrammatiko committed:
Clean up the mess

dgrammatiko committed:
Clean up the mess

dgrammatiko committed:
remove test code

dgrammatiko committed:
remove test code

Magnytu2 commented:
Hello, to support the project. I think a lot of us think you can't miss the bootstrap 5 switch now. Joomla needs to take a step ahead and not stay in the back of the train. I can understand that technically it is complicated. But strategically, choosing Bootstrap 5 would be great for the future of Joomla. I love Joomla!

kawshar commented:
+1 for BS5 (Y)

dgrammatiko commented:
Almost there: 

Bootstrap components done:
- [ x ]  alert
- [ x ]  button
- [ x ]  carousel
- [ x ]  collapse
- [ x ]  dropdown
- [ x ]  modal
- [ ]  popover
- [ x ]  scrollspy
- [ x ]  tab
- [ ]  toast
- [ ]  tooltip

Joomla PHP side:
- [ ]  toast

Joomla JS modification side:
- [ ]  Field Modals
- [ ]  Side by Side Editing com Associations
- [ ]  Other things that I haven't tested yet

dgrammatiko committed:
Add Toast support

dgrammatiko committed:
Add Toast support

dgrammatiko committed:
version correction

dgrammatiko committed:
version correction

ciar4n commented:
> The biggest break to my mind right now is two things. First the form group class is removed - they want to use mb-3 or similar going forwards - just utility classes for padding - so all our form styling is way out (possibly fixable with @extends).

The only reason the `.form-group` class was removed from BS5 was that it did nothing more than add a bottom margin. It was a duplicated class so its removal makes sense. No reason not to add it back in let's say a system.css (presuming that's still a thing)

> Second we need to refactor every media query in the system because their scss mixin changed per their docs.

I haven't examined it closely but seems like nothing more than a find n replace. Maybe I'm missing something.

ciar4n commented:
I honestly don't think the template developers will have an issue here. If they already override everything they likely will continue doing so. If they don't then the changes are minimal

If there is an overall commitment to this then I'll throw my hat in and commit to help converting the core views.

dgrammatiko committed:
tooltips + popups

dgrammatiko committed:
tooltips + popups

dgrammatiko committed:
CS

dgrammatiko committed:
CS

dgrammatiko committed:
CS?

dgrammatiko committed:
CS?

dgrammatiko committed:
CS?.

dgrammatiko committed:
CS?.

dgrammatiko committed:
CS?.#

dgrammatiko committed:
CS?.#

dgrammatiko committed:
Grrr

dgrammatiko committed:
Grrr

wilsonge committed:
Fix colour

dgrammatiko committed:
dropdowns

dgrammatiko committed:
dropdowns

wilsonge committed:
Revert some of the code to use official bootstrap js

dgrammatiko committed:
v12

dgrammatiko committed:
v12

dgrammatiko committed:
missing bs

dgrammatiko committed:
missing bs

dgrammatiko committed:
v13

dgrammatiko committed:
v13

infograf768 commented:
Will wait to test for side by side in com_associations.

simbus82 commented:
Finally an improvement that all CMS's "market audiences" can understand and use as a leverage in the choice of the CMS.
Great job @dgrammatiko! I will go to try and check if something is missing.

infograf768 commented:
Also
<img width="328" alt="Screen Shot 2021-01-13 at 12 08 52" src="https://user-images.githubusercontent.com/869724/104444521-32c39300-5598-11eb-99e1-5cfd61c74d21.png">

instead of
<img width="331" alt="Screen Shot 2021-01-13 at 12 10 23" src="https://user-images.githubusercontent.com/869724/104444653-5dade700-5598-11eb-9801-ee090daa96c4.png">

dgrammatiko commented:
@infograf768 that the 3rd bullet:

> The Toolbar dropdown buttons have wrong html the dropdown needs to be the next sibling element to the button (who did this? )

FWIW the produced HTML for the actions/etc dropdown buttons is totally wrong. It was pure lack that this was working in the first place...

infograf768 commented:
Also Multilingual Status broken (empty popup).

dgrammatiko committed:
Fix modal fields

dgrammatiko committed:
Fix modal fields

dgrammatiko commented:
@infograf768 can you please point me any breaks in com associations, I'm not familiar with the workflow. Side by side seems ok (-ish ?):
![Screenshot 2021-01-13 at 13 54 52](https://user-images.githubusercontent.com/3889375/104455188-37437800-55a7-11eb-82d7-7b70dbc27e10.png)

dgrammatiko committed:
jQuery is not needed here anymore

dgrammatiko committed:
jQuery is not needed here anymore

dgrammatiko committed:
Modal fixes

dgrammatiko committed:
Modal fixes

ITPrism commented:
I found some JavaScript errors.
I have installed Blog Sample Data which includes the workflows.

### Smart Search Statistics
There is an error when click on the button "Statistics"
![statistics](https://user-images.githubusercontent.com/1272206/104465719-040be380-55bd-11eb-8ac1-70cc67402b1c.png)

### Create or Edit an item
There is an error when open a page for editing or adding new item.
![new_edit_item](https://user-images.githubusercontent.com/1272206/104465880-3584af00-55bd-11eb-895f-fccb975209b8.png)

### Articles & Featured Articles
There is an error in admin-items-workflow-buttons.min.js on the pages Articles and Featured Articles.
![featured_articles](https://user-images.githubusercontent.com/1272206/104468133-c78db700-55bf-11eb-861f-7562e9bd9e59.png)

### Button for editing a module on the front-end
An error occurs when you place the mouse on the button.
![mouse_over_edit_button](https://user-images.githubusercontent.com/1272206/104467965-9319fb00-55bf-11eb-9f58-d4a921c2b619.png)

pinta83 commented:
+1 for BS5

dgrammatiko committed:
Modal further fixes

dgrammatiko committed:
Modal further fixes

dgrammatiko commented:
> There is an error when click on the button "Statistics"

Should be fixed

> There is an error when open a page for editing or adding new item.

Should be fixed

> There is an error in admin-items-workflow-buttons.min.js on the pages Articles and Featured Articles.

Toolbar Dropdowns are a mess

> An error occurs when you place the mouse on the button.

Noted

Quy commented:
Please fix `Copy Template`, `Manage Folders` and `New File` modals under System>SiteTemplates>Cassiopeia Details and Files.

ITPrism commented:
### Errors when click on buttons "Versions" and "Preview"
![versions_preview](https://user-images.githubusercontent.com/1272206/104485486-388a9a00-55d3-11eb-9044-bf5174d4c8d9.png)

### Errors on banners tracks when click on the button "Export" 
![banner_tracks](https://user-images.githubusercontent.com/1272206/104485695-7be50880-55d3-11eb-85ae-405a5c09c996.png)

wilsonge committed:
Add back white scss variable

wilsonge committed:
All remaining breakpoints and add back white css var

wilsonge committed:
Remove input-append and input-prepend classes

wilsonge committed:
Remove commented code

wilsonge committed:
Fix close button of modals

wilsonge committed:
Article padding cleanup

wilsonge commented:
Joomla Update is generally broken. It's unrelated to anything

wilsonge committed:
Fix stats admin module js

wilsonge committed:
Fix stats admin module js

infograf768 commented:
> @infograf768 can you please point me any breaks in com associations, I'm not familiar with the workflow. Side by side seems ok (-ish ?):

The Select Target should display a popup with the list of items to select from.
Popup is empty.

<img width="933" alt="Screen Shot 2021-01-14 at 09 32 26" src="https://user-images.githubusercontent.com/869724/104564826-ac688900-564b-11eb-9ad5-2e2db9210bbf.png">

<img width="910" alt="Screen Shot 2021-01-14 at 09 32 42" src="https://user-images.githubusercontent.com/869724/104564850-b4c0c400-564b-11eb-8ca5-8323b07b9b35.png">

Console error:
<img width="1393" alt="Screen Shot 2021-01-14 at 09 35 30" src="https://user-images.githubusercontent.com/869724/104564993-e76abc80-564b-11eb-9193-e7c8bcc79f4c.png">

infograf768 commented:
The com_associations issue looks like it is solved in this updated present branch.

ceford commented:
Question: Fontawesome icons or Bootstrap icons - any opinions?

simbus82 commented:
> Question: Fontawesome icons or Bootstrap icons - any opinions?

Bootstrap 5 has its own open source SVG icon library, could be "very clean" use these icons without load other libraries. IMHO.
But I don't know what kind of effect it might have on the admin template ...

dgrammatiko commented:
@ceford @simbus82 please take this icons conversation to the discussion thread https://github.com/joomla/joomla-cms/discussions/3176 , let's keep this JS related only. Thanks

dgrammatiko committed:
attributes

dgrammatiko committed:
Merge branch '4.0-dev__BS5' of github.com:dgrammatiko/joomla-cms into 4.0-dev__BS5

dgrammatiko committed:
oops

Quy commented on administrator/components/com_installer/tmpl/update/default.php:
```diff
@@ -105,7 +105,7 @@
 											<span class="hasPopover"
 												  title="<?php echo Text::_('COM_INSTALLER_DOWNLOADKEY_MISSING_LABEL') ?>"
-												  data-content="<?php echo Text::_('COM_INSTALLER_DOWNLOADKEY_MISSING_TIP') ?>"
+												  data-bs-contentt="<?php echo Text::_('COM_INSTALLER_DOWNLOADKEY_MISSING_TIP') ?>"
```
```suggestion
												  data-bs-content="<?php echo Text::_('COM_INSTALLER_DOWNLOADKEY_MISSING_TIP') ?>"
```

Quy commented on administrator/components/com_modules/tmpl/module/edit_assignment.php:
```diff
@@ -133,7 +133,7 @@
 					<div class="nav-hover treeselect-menu">
 						<div class="dropdown">
-							<button type="button" data-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-light">
+							<button type="button"  data-bs-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-light">
```
```suggestion
							<button type="button" data-bs-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-light">
```

Quy commented on administrator/components/com_mails/tmpl/templates/default.php:
```diff
@@ -64,7 +64,7 @@
 								<td class="break-word">
 									<div class="dropdown">
-										<a class="dropdown-toggle" href="#" role="button" id="mTemplate<?php echo $i; ?>" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+										<a class="dropdown-toggle" href="#" role="button" id="mTemplate<?php echo $i; ?>"  data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```
```suggestion
										<a class="dropdown-toggle" href="#" role="button" id="mTemplate<?php echo $i; ?>" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```

dgrammatiko committed:
Update administrator/components/com_mails/tmpl/templates/default.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented on administrator/components/com_menus/tmpl/menus/default.php:
```diff
@@ -151,15 +151,15 @@
 									<?php if (isset($this->modules[$item->menutype])) : ?>
 										<div class="dropdown">
-											<button type="button" class="btn btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown">
+											<button type="button" class="btn btn-secondary btn-sm dropdown-toggle"  data-bs-toggle="dropdown">
```
```suggestion
											<button type="button" class="btn btn-secondary btn-sm dropdown-toggle" data-bs-toggle="dropdown">
```

Quy commented on administrator/components/com_users/tmpl/users/default.php:
```diff
@@ -118,7 +118,7 @@
 										<?php echo HTMLHelper::_('users.addNote', $item->id); ?>
 										<?php if ($item->note_count > 0) : ?>
-										<button type="button" class="btn btn-secondary btn-sm dropdown-toggle dropdown-toggle-split" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+										<button type="button" class="btn btn-secondary btn-sm dropdown-toggle dropdown-toggle-split"  data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```
```suggestion
										<button type="button" class="btn btn-secondary btn-sm dropdown-toggle dropdown-toggle-split" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
```

dgrammatiko committed:
Update administrator/components/com_menus/tmpl/menus/default.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented on administrator/modules/mod_user/tmpl/default.php:
```diff
@@ -15,13 +15,13 @@
 ?>
 <div class="header-item-content dropdown header-profile d-flex">
-	<button class="dropdown-toggle d-flex flex-column align-items-stretch <?php echo ($hideLinks ? 'disabled' : ''); ?>" data-toggle="dropdown" type="button" <?php echo ($hideLinks ? 'disabled' : ''); ?>
+	<button class="dropdown-toggle d-flex flex-column align-items-stretch <?php echo ($hideLinks ? 'disabled' : ''); ?>"  data-bs-toggle="dropdown" type="button" <?php echo ($hideLinks ? 'disabled' : ''); ?>
```
```suggestion
	<button class="dropdown-toggle d-flex flex-column align-items-stretch <?php echo ($hideLinks ? 'disabled' : ''); ?>" data-bs-toggle="dropdown" type="button" <?php echo ($hideLinks ? 'disabled' : ''); ?>
```

dgrammatiko committed:
Update administrator/components/com_users/tmpl/users/default.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Update administrator/modules/mod_user/tmpl/default.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Update administrator/components/com_installer/tmpl/update/default.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Update administrator/components/com_modules/tmpl/module/edit_assignment.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented on administrator/templates/atum/html/layouts/chromes/body.php:
```diff
@@ -45,7 +45,7 @@
 			<?php $dropdownPosition = Factory::getLanguage()->isRTL() ? 'left' : 'right'; ?>
 			<div class="module-actions dropdown">
-				<button type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
+				<button type="button"  data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
```
```suggestion
				<button type="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
```

dgrammatiko committed:
Update administrator/templates/atum/html/layouts/chromes/body.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented on administrator/templates/atum/html/layouts/chromes/well.php:
```diff
@@ -50,7 +50,7 @@
 					<?php $dropdownPosition = Factory::getLanguage()->isRTL() ? 'left' : 'right'; ?>
 					<div class="module-actions dropdown">
-						<button type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
+						<button type="button"  data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
```
```suggestion
						<button type="button" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="btn" id="dropdownMenuButton-<?php echo $id; ?>">
```

Quy commented on libraries/src/HTML/Helpers/ActionsDropdown.php:
```diff
@@ -40,7 +40,7 @@ public static function render($item = '')
 		$html = array();
 
-		$html[] = '<button data-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-secondary">';
+		$html[] = '<button  data-bs-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-secondary">';
```
```suggestion
		$html[] = '<button data-bs-toggle="dropdown" class="dropdown-toggle btn btn-sm btn-secondary">';
```

Quy commented on libraries/src/Form/Field/ModulepositionField.php:
```diff
@@ -155,7 +155,7 @@ protected function getInput()
 		$html[] = parent::getInput()
 			. '<a class="btn" title="' . Text::_('COM_MODULES_CHANGE_POSITION_TITLE') . '"  href="' . $link
-			. '"  data-toggle="modal" data-target="#modulePositionModal">'
+			. '"  data-bs-toggle="modal" data-bs-target="#modulePositionModal">'
```
```suggestion
			. '" data-bs-toggle="modal" data-bs-target="#modulePositionModal">'
```

dgrammatiko committed:
Update administrator/templates/atum/html/layouts/chromes/well.php

Co-authored-by: Quy <quy@fluxbb.org>

Quy commented on libraries/src/HTML/Helpers/Dropdown.php:
```diff
@@ -95,7 +95,7 @@ public static function start()
 
 		$dropDownList = '<div class="btn-group" style="margin-left:6px;display:none">
-							<a href="#" data-toggle="dropdown" class="dropdown-toggle btn btn-secondary btn-sm">
+							<a href="#"  data-bs-toggle="dropdown" class="dropdown-toggle btn btn-secondary btn-sm">
```
```suggestion
							<a href="#" data-bs-toggle="dropdown" class="dropdown-toggle btn btn-secondary btn-sm">
```

dgrammatiko committed:
Update libraries/src/Form/Field/ModulepositionField.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Update libraries/src/HTML/Helpers/Dropdown.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
Update libraries/src/HTML/Helpers/ActionsDropdown.php

Co-authored-by: Quy <quy@fluxbb.org>

dgrammatiko committed:
clean up 1

dgrammatiko committed:
Merge branch '4.0-dev__BS5' of github.com:dgrammatiko/joomla-cms into 4.0-dev__BS5

dgrammatiko committed:
Fix the init methods

dgrammatiko committed:
Toolbars take #1

dgrammatiko committed:
Fix Modals console logs

dgrammatiko committed:
More modal fixes

dgrammatiko commented:
> Also Multilingual Status broken (empty popup).

@infograf768 this should be fixed now

dgrammatiko committed:
Banners Modal

dgrammatiko committed:
Better default selector

dgrammatiko committed:
CS

dgrammatiko committed:
Element fuse

wilsonge committed:
Remove btn-block class

wilsonge committed:
Add back some spacing to card columns

wilsonge committed:
Replace float-left and float-right

wilsonge committed:
Revert back layouts to original locations

chnnst commented:
@dgrammatiko what still we need to test for Bootstrap 5 #31990 ?

micker commented:
> @dgrammatiko what still we need to test for Bootstrap 5 #31990 ?

yes ready to help

ITPrism commented:
There is an error when we click on the button "Actions".
![actions](https://user-images.githubusercontent.com/1272206/104704930-f2901c00-5721-11eb-91a9-fd80ab2e31a5.png)

dgrammatiko commented:
> There is an error when we click on the button "Actions".

Thanks @ITPrism this is a known problem (many toolbar buttons are still broken)

dgrammatiko commented:
> @dgrammatiko what still we need to test for Bootstrap 5 #31990 ?

@chnnst @micker please try all the components frontend/backend. Basically we need to test every possible view

micker commented:
hello i check many many thing in backend all seems works no error in console
- adding/modify menu ok
- adding/modify item ok
- adding/modify category ok
- testing banner ok
- testing smart search ok
- testing tag ok
- testing user ok
- testing system component (checking ...)
- testing language ok
- testing custom atum
- impressiv works... all works no change (that good !)

micker commented:
mobile admin works good

ITPrism commented:
The popovers on the page Articles are broken.

### Expected result
![after_popover](https://user-images.githubusercontent.com/1272206/104742284-ea9ca000-5752-11eb-83ca-c9fc82b9c477.png)

### Actual result
![actual_popover](https://user-images.githubusercontent.com/1272206/104742340-f9835280-5752-11eb-9d29-352d2d642a7d.png)

micker commented:
Testing front office
- dropdown menu ok
- responsive ok
- module edition ok
- item edition ok
- backtotop ok
- versionning ok

micker commented:
> The popovers on the page Articles are broken.
> 
> ### Expected result
> ![after_popover](https://user-images.githubusercontent.com/1272206/104742284-ea9ca000-5752-11eb-83ca-c9fc82b9c477.png)
> 
> ### Actual result
> ![actual_popover](https://user-images.githubusercontent.com/1272206/104742340-f9835280-5752-11eb-9d29-352d2d642a7d.png)

for me its ok
![image](https://user-images.githubusercontent.com/2539440/104742578-e53b5780-574a-11eb-819f-94b2a455a539.png)

ITPrism commented:
@micker Are you sure you are testing the right branch?!? :)

micker commented:
I install bs5 component over a fresh install on beta7

dgrammatiko committed:
Workflow popovers

dgrammatiko commented:
@ITPrism workflow popups fixed 
![Screenshot 2021-01-15 at 16 56 09](https://user-images.githubusercontent.com/3889375/104749868-9b0aa400-5753-11eb-8a0f-afa5deca498d.png)

dgrammatiko committed:
CS

dgrammatiko committed:
popover on hover

dgrammatiko committed:
undo popup hover

dgrammatiko commented:
@micker I guess you have to download the version from the Github PR at the bottom there's a download (with a green tick) you have to click on the details to take you to the downloads page
![Screenshot 2021-01-15 at 17 35 00](https://user-images.githubusercontent.com/3889375/104753419-3dc52180-5758-11eb-8208-6873f5b5f03a.png)

micker commented:
i will check i can't just update the component ?

todordev committed:
Fixed some styles on the last step of the installation process. (#51)

Co-authored-by: Todor <todor@funfex.com>

wilsonge committed:
Spaces to tabs

wilsonge committed:
SCSS Lint

wilsonge committed:
Fix JS Codestyle

wilsonge committed:
Improved fix to toolbar dropdowns

wilsonge committed:
Comments for the changes

wilsonge committed:
Spaces to tabs

wilsonge committed:
Various fixes

wilsonge committed:
de -> en

wilsonge committed:
Try and fix tests in a dirty way

wilsonge committed:
Move sr-only to visually-hidden

wilsonge committed:
Spaces to tabs

wilsonge committed:
Rename files back to original versions

wilsonge committed:
SCSS Lint

wilsonge committed:
Missing helpers in atum

wilsonge committed:
Remove test property

PhilETaylor committed:
s/Unexising/Unexisting (#32048)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
fix Notice: Only variables should be passed by reference (#32052)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
Fix default value type does not match the given type (#32055)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
add missing include exception (#32057)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
[4] add missing and fix wrong includes in com_finder, and remove unset on undefined var (#32058)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

PhilETaylor committed:
Remove unset call as $xml is not set when here (#32054)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Use @wilsonege ’s dropdown fixes (cleaner)

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into 4.0-dev__BS5

wilsonge committed:
Update administrator/templates/atum/scss/vendor/_bootstrap.scss

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Update administrator/components/com_menus/tmpl/menus/default.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Update administrator/templates/atum/html/layouts/chromes/body.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Update administrator/templates/atum/html/layouts/chromes/well.php

Co-authored-by: Quy <quy@fluxbb.org>

anibalsanchez commented:
I've built the release from the branch dgrammatiko:4.0-dev__BS5, installed it from scratch and test the general management of a site. It is working in the same way than the latest J4 b6 and I couldn't find bugs. :+1:

PhilETaylor committed:
add missing import for typehint in docblock (#32050)

Signed-off-by: Phil E. Taylor <phil@phil-taylor.com>

richard67 commented on layouts/joomla/toolbar/basic.php:
```diff
@@ -10,11 +10,12 @@
 Factory::getDocument()->getWebAssetManager()
-	->useScript('core')
-	->useScript('webcomponent.toolbar-button');
+		->useScript('core')
+		->useScript('webcomponent.toolbar-button');
```
```suggestion
	->useScript('core')
	->useScript('webcomponent.toolbar-button');
```
Fix PHP CS for Drone.

dgrammatiko committed:
Update layouts/joomla/toolbar/basic.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

richard67 commented:
@dgrammatiko Now Drone is happy with PHP CS, but it reports javascript linter errors here [https://ci.joomla.org/joomla/joomla-cms/39245/1/21](https://ci.joomla.org/joomla/joomla-cms/39245/1/21):

```
/********/src/build/media_source/com_menus/js/admin-menus-default.es6.js
  20:65  error  Array.prototype.map() expects a return value from arrow function  array-callback-return
```

C-Lodder committed:
Update default.php (#32067)

dgrammatiko committed:
CS js

dgrammatiko committed:
Update for the latest changes

- Fixed Events forced to use the Browsers native javascript, no B/C break anymore

dgrammatiko committed:
Fix the template modals

dgrammatiko commented:
> Please fix Copy Template, Manage Folders and New File modals under System>SiteTemplates>Cassiopeia Details and Files.

Fixed

dgrammatiko commented:
One note about Frontend editing the popup is triggered (although misplaced probably due to CSS differences between V4 and V5) so I will take that as a fixed atm. Also @brianteeman is refactoring the code to be plain css so it really doesn't make any sense to spent time on this.
![Screenshot 2021-01-17 at 14 21 06](https://user-images.githubusercontent.com/3889375/104844283-b1d60580-58cf-11eb-9f58-571a09217d4b.png)

So, atm all the reported issues have been fixed

brianteeman commented:
That popup is already css only and already misplaced before this pr

dgrammatiko commented:
> That popup is already css only and already misplaced before this pr

That's weird, the code initialises a popup here:
https://github.com/joomla/joomla-cms/blob/0f955c88920d650e026b987732d49056d3e0be39/build/media_source/legacy/js/frontediting.es5.js#L64

brianteeman commented:
That's the popover for the module and menu links
The edit article link has never used that script

dgrammatiko commented:
> That's the popover for the module and menu links

I'm an idiot, testing the wrong thing here 🤦

wilsonge committed:
Fix modal not showing

dgrammatiko committed:
ml renamed to me (right -> end)

wilsonge committed:
Update administrator/components/com_postinstall/tmpl/messages/default.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Margin classes

wilsonge committed:
Backport sr-only class

wilsonge committed:
Remove comment

ciar4n committed:
Badge backgrounds (#52)

HLeithner committed:
cs

wilsonge committed:
Update modules/mod_stats/tmpl/default.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Duplicate class names for puublished field

Quy commented:
> > Please fix Copy Template, Manage Folders and New File modals under System>SiteTemplates>Cassiopeia Details and Files.
> 
> Fixed, @Quy can you confirm it?

Confirmed fix. Thanks.

wilsonge committed:
Text-left and text-right

wilsonge committed:
Fix error in system view

dgrammatiko committed:
Front end menu needs collapse

dgrammatiko committed:
mod_languages has a dropdown

wilsonge committed:
Fix range form field class

wilsonge committed:
Fix JS error

wilsonge committed:
Fix close button on modal

richard67 commented:
@dgrammatiko Are you sure that the frontend metismenu needs bootstrap collapse?

micker commented:
Hello just question de need to test 2 pr for bs5 or just this ?

dgrammatiko committed:
Template is WRONGLY opinionated to have the menu ALWAYS collapsible on mobiles

dgrammatiko commented:
> Are you sure that the frontend metismenu needs bootstrap collapse?

So, it turns out I did this wrong. Both the default and the dropdown (metismenu) are becoming collapsible (burger icon) in the template. FWIW this is extremely opinionated as a menu with very few, one level menu items could just be using the flex css (eg always visible). The part that is controlling this is hardcoded in the template (you cannot do anything about unless you clone the template and remove the hardcoded parts there). Anyways this is just a note as I revisited the burger menu interactivity (it's on the template level not on the menu layouts as I thought). Hope this explanation makes sense, not enough coffee yet...

richard67 commented:
> So, it turns out I did this wrong. Both the default and the dropdown (metismenu) are becoming collapsible (burger icon) in the template. FWIW this is extremely opinionated as a menu with very few, one level menu items could just be using the flex css (eg always visible). The part that is controlling this is hardcoded in the template (you cannot do anything about unless you clone the template and remove the hardcoded parts there).

So it was good that I've asked. Do you think we should (with another PR of course) fix the template to be more flexible at this point?
 
> @micker I really have no clue but I understand that the production team needs to decide first and then it's up to @wilsonge how that's gonna be handled

I think he just wanted to know if he can test the 2 PR separately, or if they should be tested together.

micker commented:
@richard67 yes that my question sorry for my bad english lol

dgrammatiko commented:
> So it was good that I've asked. Do you think we should (with another PR of course) fix the template to be more flexible at this point?

Yup, good catch. I think wrapping the dropdown into an if/else on a template param (a simple switch defaults to true, always burger menu for b/C) it's not a lot of work. The reason is that maybe people have a simple site with very few menu items so the js behaviour could be useless in that case (a JS guy evangelising no-js solution 😂)

@micker so the 2 PRs are doing different things, George's PR is all about the visual consistency, this PR is all about the JS driven interactivity. So in that sense both PRs need some extensible testing. I think if there's a positive decision on these then maybe merging them and then releasing a Beta so people could test it more broadly is a good way forward (but I'm not making the rules here 😉)

wilsonge committed:
Dropdown menu and btn close

ciar4n committed:
Use .form-select (#55)

Hackwar committed:
Adding delay in toggleSendMail test (#56)

Hackwar committed:
Use alternative method to click radio button switcher (#57)

wilsonge committed:
Update libraries/src/HTML/Helpers/Grid.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Update libraries/src/HTML/Helpers/Grid.php

Co-authored-by: Quy <quy@fluxbb.org>

wilsonge committed:
Fix input

wilsonge committed:
Fix button group styling

dgrammatiko committed:
language overrides results disapear

ciar4n committed:
J4 table styling

ciar4n committed:
Update login.php

ciar4n committed:
Update _variables.scss

ciar4n committed:
Update joomla-image-select.css

dgrammatiko committed:
Fix language overrides missing results (#58)

chnnst commented:
So hat's broken now?

Only Frontend Editing is not fixed, or enything else we should test?

dgrammatiko commented:
> What is still in WIP ?

ATM all the reported bugs are fixed (apart the frontend editing) but you're welcome to test yourself and report back if something is broken

Magnytu2 commented:
@dgrammatiko I'm not a developer, and Github remains a big mystery to me. How to try and test it all? There are several github pages and I doubt each time. # 31990 # 32037 # 31765

anibalsanchez commented:
@Magnytu2 These links are informative about how to test Joomla issues:

https://docs.joomla.org/Testing_Joomla!_patches
https://docs.joomla.org/Portal:Bug_Squad

In particular, for this issue, I tried to use the [Patch Tester](https://docs.joomla.org/Testing_Joomla!_patches#Download_Patch_Tester), but it didn't apply it. So, I downloaded the ZIP file for this branch of the @dgrammatiko repo and build it to test it. https://docs.joomla.org/Special:MyLanguage/J4.x:Setting_Up_Your_Local_Environment

Magnytu2 commented:
So this is not enough?
![Capture d’écran 2021-01-21 à 10 41 12](https://user-images.githubusercontent.com/26197929/105332846-578dbb00-5bd5-11eb-8386-947d9edea671.png)

dgrammatiko commented:
> So this is not enough?

Sure just make sure you're using the Full package

dgrammatiko committed:
Fix the finder accordion

ITPrism commented:
# Frontend

## There is JavaScript error when you place the cursor on a link .
![tootlip_errors](https://user-images.githubusercontent.com/1272206/105337389-05539680-5be3-11eb-9969-10b2c3924952.png)
---
## JS error when you click on the button Versions.
![versions](https://user-images.githubusercontent.com/1272206/105337505-2ddb9080-5be3-11eb-887f-94a038816096.png)
---
## The tooltip should be over the label "Unpublished" and the icon next to the link "Edit" should be "_icon-edit_".
![icon](https://user-images.githubusercontent.com/1272206/105337766-6ed3a500-5be3-11eb-9f45-4fa6fec75575.png)
---
# Backend
## JS error when you click on the buttons "Save and Close" on the view "Edit Article".
![save_click](https://user-images.githubusercontent.com/1272206/105337857-8ca10a00-5be3-11eb-8fb7-6afa462357f1.png)
---
## The buttons for changing states of an article to published and featured do not work.
![buttons_featured_publish](https://user-images.githubusercontent.com/1272206/105338035-bf4b0280-5be3-11eb-9bd8-b3e5f9558c97.png)

richard67 commented:
@dgrammatiko The system tests (mysql test, but this just because it comes first, the problem is not related to the database) systematically fail now at the same place when trying to delete a custom field with following error message:
```
1) ContentFieldListCest: Field
 Test  tests/Codeception/acceptance/administrator/components/com_content/ContentFieldListCest.php:Field
                                                                                                                                                  
  [Facebook\WebDriver\Exception\JavascriptErrorException] javascript error: jQuery is not defined
```
see log here: [https://ci.joomla.org/joomla/joomla-cms/39376/1/22](https://ci.joomla.org/joomla/joomla-cms/39376/1/22).

Maybe it needs to make a similar adjustment to the tests as @wilsonge has done in his PR #32037 for the tests? Or maybe some jQuery dependency left?

Magnytu2 commented:
If I'm not mistaken, bootstrap 5 allows for a different layout of containers. With a maximum width of 1320px which is more suitable for accessibility. I cannot find this width on my local test in Cassiopeia.
![Capture d’écran 2021-01-21 à 11 31 47](https://user-images.githubusercontent.com/26197929/105339494-041f6b00-5bdd-11eb-8efd-a863dee2b40b.png)

ITPrism commented:
### It does not add tag to mail template body.
![email_templates](https://user-images.githubusercontent.com/1272206/105343204-fd4b2500-5be9-11eb-8881-caca8bf5e7dd.png)

dgrammatiko commented:
> It does not add tag to mail template body.

@ITPrism fixed https://github.com/joomla/joomla-cms/pull/31990/commits/948998f770c48ffa311563e2f66e7cf5fcce63c4

Commit:
Fix input

Commit:
Fix input

dgrammatiko committed:
Fix versions Modal FE

dgrammatiko commented:
> JS error when you click on the button Versions.

@ITPrism  fixed with https://github.com/joomla/joomla-cms/pull/31990/commits/973849a2e44d0c66764bc3ce15e4aeb24089cc74

dgrammatiko committed:
Fix frontend editing tooltips

Please merge Brian’s PR

dgrammatiko commented:
> There is JavaScript error when you place the cursor on a link .

@ITPrism fixed with https://github.com/joomla/joomla-cms/pull/31990/commits/05f454e0c724cd8fc368d93f5aa2dd0130a8b648


![Screenshot 2021-01-21 at 19 55 10](https://user-images.githubusercontent.com/3889375/105398602-d4935180-5c22-11eb-9f37-fe941bca5e9a.png)


A note here: @brianteeman 's PR that eliminates the need for JS is way better but I did this just for the shake of consistency (everything works as before)

brianteeman commented:
Thanks for the compliment

wilsonge committed:
Fix accordions

wilsonge committed:
Merge branch '4.0-dev' into bs5

dgrammatiko committed:
mail templates/ frontediting

dgrammatiko committed:
toolbar buttons

dgrammatiko committed:
Rebase to George’s branch

# Conflicts:
#	administrator/components/com_admin/tmpl/help/default.php
#	administrator/components/com_mails/tmpl/templates/default.php
#	administrator/components/com_menus/src/Field/MenutypeField.php
#	administrator/components/com_menus/tmpl/menus/default.php
#	administrator/components/com_modules/tmpl/module/edit_assignment.php
#	administrator/components/com_templates/src/Service/HTML/Templates.php
#	administrator/components/com_users/tmpl/users/default.php
#	administrator/modules/mod_feed/tmpl/default.php
#	administrator/modules/mod_latest/tmpl/default.php
#	administrator/modules/mod_latestactions/tmpl/default.php
#	administrator/modules/mod_logged/tmpl/default.php
#	administrator/modules/mod_popular/tmpl/default.php
#	administrator/modules/mod_user/tmpl/default.php
#	administrator/templates/atum/html/layouts/chromes/body.php
#	administrator/templates/atum/html/layouts/chromes/well.php
#	build/media_source/com_menus/js/admin-menus-default.es6.js
#	build/media_source/system/js/fields/joomla-field-media.w-c.es6.js
#	build/media_source/system/js/fields/joomla-field-user.w-c.es6.js
#	components/com_contact/tmpl/categories/default_items.php
#	components/com_content/tmpl/categories/default_items.php
#	components/com_content/tmpl/category/blog_children.php
#	components/com_content/tmpl/category/default_children.php
#	components/com_newsfeeds/tmpl/categories/default_items.php
#	layouts/joomla/button/action-button.php
#	layouts/joomla/button/transition-button.php
#	layouts/joomla/form/field/contenthistory.php
#	layouts/joomla/toolbar/basic.php
#	layouts/joomla/toolbar/dropdown.php
#	layouts/joomla/toolbar/modal.php
#	layouts/joomla/toolbar/versions.php
#	layouts/libraries/html/bootstrap/modal/header.php
#	libraries/src/Form/Field/ModulepositionField.php
#	libraries/src/HTML/Helpers/ActionsDropdown.php
#	libraries/src/HTML/Helpers/Bootstrap.php
#	libraries/src/HTML/Helpers/Dropdown.php
#	libraries/src/HTML/Helpers/Grid.php
#	libraries/src/Toolbar/Button/PopupButton.php

dgrammatiko committed:
Mail templates

wilsonge committed:
Fix margins on media manager file tree

wilsonge committed:
Fix some table styling

wilsonge committed:
Fix border issue on hovered vertical tabs

wilsonge committed:
Whoops

wilsonge committed:
Fix modal padding on header

wilsonge committed:
More com_media tweaks

dgrammatiko committed:
JS part

dgrammatiko committed:
Merge branch '__george' into __BS5_all

dgrammatiko committed:
PHP CS

wilsonge committed:
Fix data attributes for launching modals

dgrammatiko committed:
Merge branch 'bs5' of github.com:wilsonge/joomla-cms into __george

dgrammatiko committed:
missing npm dependencies

dgrammatiko committed:
Grrr who designed this system? Wait, I did…

wilsonge committed:
Missing dependency on bootstrap framework

dgrammatiko committed:
Merge branch 'bs5' of github.com:wilsonge/joomla-cms into __george

dgrammatiko committed:
Merge branch '__george' into __BS5_all

dgrammatiko committed:
collapse

wilsonge committed:
Fix pagination dropdown to be a slect

dgrammatiko committed:
Merge branch 'bs5' of github.com:wilsonge/joomla-cms into __george

dgrammatiko committed:
Merge branch '__george' into __BS5_all

dgrammatiko committed:
CS

wilsonge committed:
Play with form-inline to backport - but also remove some core instances

dgrammatiko committed:
meh the directory is wrong

dgrammatiko committed:
Merge branch 'bs5' of github.com:wilsonge/joomla-cms into __george

dgrammatiko committed:
Merge branch '__george' into __BS5_all

dgrammatiko committed:
Exclude bootstrap from the usual comile js dance

dgrammatiko committed:
Javascript fun

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into __BS5_all

# Conflicts:
#	administrator/components/com_media/resources/scripts/components/modals/modal.vue
#	administrator/components/com_modules/tmpl/select/default.php
#	build/media_source/com_menus/js/admin-menus-default.es6.js
#	build/media_source/system/js/fields/joomla-field-media.w-c.es6.js
#	build/media_source/system/js/fields/joomla-field-user.w-c.es6.js
#	components/com_finder/tmpl/search/default_form.php
#	layouts/joomla/button/action-button.php
#	layouts/joomla/form/field/contenthistory.php
#	layouts/joomla/toolbar/dropdown.php
#	libraries/src/HTML/Helpers/Bootstrap.php
#	libraries/src/Toolbar/Button/PopupButton.php
#	templates/cassiopeia/error.php
#	templates/cassiopeia/index.php

dgrammatiko committed:
CS, etc

dgrammatiko committed:
CS

dgrammatiko committed:
cs

dgrammatiko committed:
Remove files

dgrammatiko committed:
cs

dgrammatiko committed:
oops

dgrammatiko committed:
Restore Bootstrap.framework for B/C

dgrammatiko committed:
Undo the docblock

dgrammatiko committed:
CS

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into __BS5_all

dgrammatiko committed:
remove frontediting js

dgrammatiko committed:
oops the filename is wrong

dgrammatiko committed:
Csssssss

dgrammatiko committed:
The workflow targeted through the parent button

infograf768 commented:
Restarted drone.

infograf768 commented:
@dgrammatiko 
please see https://github.com/joomla/joomla-cms/pull/32101#issuecomment-765880351

1. It looks like bootstrap5 rtl is not loaded.
2. I have no idea where that css comes from as I don't see an equivalent in scss.

infograf768 commented:
Tried to apply this patch on 4.-dev
I have an error

```
> joomla@4.0.0 build:bs5 /Applications/MAMP/htdocs/newfolder/joomla40
> node build/build-modules-js/build-bootstrap-js.es6.js

Building ES6 Components...
(node:4181) ExperimentalWarning: The fs.promises API is experimental
TypeError: rm is not a function
    at /Applications/MAMP/htdocs/newfolder/joomla40/build/build-modules-js/build-bootstrap-js.es6.js:122:11
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! joomla@4.0.0 build:bs5: `node build/build-modules-js/build-bootstrap-js.es6.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the joomla@4.0.0 build:bs5 script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mac/.npm/_logs/2021-01-23T08_36_48_733Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! joomla@4.0.0 install: `node build/build.js --copy-assets && node build/build.js --build-pages && npm run build:bs5`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the joomla@4.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

infograf768 commented:
line is
`await rm(resolve(outputFolder, 'index.es6.js'));`

dgrammatiko commented:
@infograf768 what's you `node -v` and `npm -v` ? Probably you need to get a newser version

infograf768 commented:
joomla40 mac$ node -v
v10.15.3
imac-2:joomla40 mac$ npm -v
6.4.1

dgrammatiko commented:
Can you d/l the current version: https://nodejs.org/dist/v14.15.4/node-v14.15.4.pkg 
FWIW it should work also with v12

infograf768 commented:
> Can you d/l the current version: https://nodejs.org/dist/v14.15.4/node-v14.15.4.pkg

That worked, thanks.

dgrammatiko committed:
Tests fixes

dgrammatiko committed:
Tests #2

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into __BS5_all

# Conflicts:
#	build/media_source/legacy/js/bootstrap-init.es5.js

dgrammatiko committed:
some clean up

dgrammatiko committed:
well null is not itteratable

dgrammatiko commented:
@brianteeman can you confirm that `npm install` in this branch works fine on your side (windows)?

brianteeman commented:
I can confirm that there were no errors reported when running `npm i`  on windows

wilsonge commented on build/media_source/system/js/searchtools.es6.js:
```diff
@@ -249,28 +249,23 @@ Joomla = window.Joomla || {};
           i.value = '';
           self.checkFilter(i);
-
-          if (window.jQuery && window.jQuery.chosen) {
```
I think we may still need this for 3rd parties who are using chosen?

dgrammatiko replied:
I'll revert those then

Fedik replied:
@wilsonge this code never worked, and will not, because there no such thing like `window.jQuery.chosen`, even if you have chose.js attached

frostmakk commented:
Open Manage Extensions.
Try to sort by clicking on the column description.
Nothing happens.

dgrammatiko committed:
revert the Chozen parts

dgrammatiko commented:
> Try to sort by clicking on the column description.

Sorry, that was me trying to fix the tests. Should be fine with:
https://github.com/joomla/joomla-cms/pull/31990/commits/12290a803deff483b12296dd831d78346eeec500

Fedik commented on build/media_source/system/js/searchtools.es6.js:
```diff
@@ -249,28 +249,23 @@ Joomla = window.Joomla || {};
-          if (window.jQuery && window.jQuery.chosen) {
-            window.jQuery(i).trigger('chosen:updated');
-          }
+          i.dispatchEvent(new Event('change'));
```
holy moly, who made it? bug on the bug, and bug behind :smile:

Fedik replied:
well, nevermind, not related to current pr, just noticed

dgrammatiko replied:
Well, it's either you or me to be blamed here 😄

frostmakk commented:
Something broken in Joomla Update.
The Pre-Update tab is missing, and Reinstall Joomla core files is missing in Live Update.
Console error: Uncaught ReferenceError: $ is not defined

dgrammatiko committed:
revert search tools add missing jQuery

dgrammatiko committed:
CS

dgrammatiko committed:
revert codeception changes

wilsonge commented:
You can ignore joomla update - it’s been broken for a few weeks. It’s unrelated to this PR - but is something I need to look into probably with @dgrammatiko

frostmakk commented:
But it seems ok in the latest nightly build.
?

dgrammatiko commented:
@wilsonge I had a PR for that https://github.com/joomla/joomla-cms/pull/18912
I'll clean things up and do a PR

frostmakk commented:
Open Options for whatever.
Console error: Uncaught TypeError: items is null
![image](https://user-images.githubusercontent.com/25645600/105607244-fc5af480-5d9d-11eb-8a30-59ac86c2e13d.png)

dgrammatiko committed:
- Add some comments
- Checks for dispose
- Checks that we have an array with elements
- Add the missing options for scrollspy

dgrammatiko commented:
> Console error: Uncaught TypeError: items is null

Fixed with https://github.com/joomla/joomla-cms/pull/31990/commits/43e64a2faf0ef55e420f1d3e27900df58336a103

Thanks @frostmakk

astridx commented:
i just run (after https://github.com/joomla/joomla-cms/pull/31990/commits/43e64a2faf0ef55e420f1d3e27900df58336a103) `git pull origin pull/31990/head:4.0-dev__BS5`, `composer i` and `npm i`.
After that that I created an contact. While saving the contact, i get the error

```
Uncaught TypeError: this._menu is undefined
    show http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:129
    toggle http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:125
    _addEventListeners http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:233
    handler http://localhost/joomla-cms4/media/vendor/bootstrap/js/dom-8eef6b5f.js:355
    addHandler http://localhost/joomla-cms4/media/vendor/bootstrap/js/dom-8eef6b5f.js:449
    on http://localhost/joomla-cms4/media/vendor/bootstrap/js/dom-8eef6b5f.js:477
    _addEventListeners http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:230
    Dropdown http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:93
    Dropdown http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:511
    <anonymous> http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:532
    <anonymous> http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:532
    <anonymous> http://localhost/joomla-cms4/media/vendor/bootstrap/js/dropdown.es6.js?dc7caea91fbb3738e28bbb84ba49f1d7:523

```

dgrammatiko committed:
Fix the console error for dropdowns

dgrammatiko committed:
Merge branch '4.0-dev' of github.com:joomla/joomla-cms into __BS5_all

# Conflicts:
#	package-lock.json
#	package.json

dgrammatiko commented:
> After that I created an contact. While saving the contact, i get the error

@astridx should been fixed with https://github.com/joomla/joomla-cms/pull/31990/commits/e62092208e844a4b52067042e49fc3ae335d555e

Thanks

dgrammatiko committed:
Fix modal initialisation for subform fields

wilsonge committed:
Comment about security

wilsonge committed:
JSCS

wilsonge committed:
JSCS

wilsonge committed:
JSCS

astridx commented:
@dgrammatiko 
> @astridx should been fixed with [e620922](https://github.com/joomla/joomla-cms/commit/e62092208e844a4b52067042e49fc3ae335d555e)
> 

Yes, now it is fine. Thanks.

astridx commented:
@dgrammatiko 
> @astridx should been fixed with [e620922](https://github.com/joomla/joomla-cms/commit/e62092208e844a4b52067042e49fc3ae335d555e)
> 

Yes, now it is fine. Thanks.

wilsonge commented:
I think this is good to go to get testing going. Thankyou very much @dgrammatiko !

brianteeman commented:
woohoo - thanks
